### ISSUE TITLE
BUG(?): [XCDYouTubeVideoOperation handleJavaScriptPlayerResponse] blocks UI

### ISSUE THREAD

**Issue by AMI289** [0xced/XCDYouTubeKit#147]
Hey guys,

First I want to say thank you for this awesome library!

Second,
I'm having a weird issue where it seems that using XCDYouTubeKit blocks the UI.

I'm new to Xcode's Instruments, but running the 'Time Profiler' helped me narrow it down to the method in the title.

I'm only using the `[[XCDYouTubeClient defaultClient] getVideoWithIdentifier:_videoID completionHandler:^(XCDYouTubeVideo *video, NSError *error)` method.

The method in the title seems to be running on the main queue, even if I explicitly create a background queue and run the above method on it.

Linking / Un-linking the JavaScriptCore framework to my project does not seem to make any difference.

I'm making a youtube query, with 25 results, then I call the method above to create XCDYouTubeVideo object for each of the results.
Perhaps the XCDYouTubeKit is too resource intense to be running this much simultaneously?

I'm running the app on iPhone 4s with iOS 7.1.2
And using Xcode 6.2 (The latest version compatible with Mavericks)

Any one else experiencing the same issue?

Basically, I'm creating my own objects for each youtube search result,
I'm just using the XCDYouTubeKit library to retrieve the stream URLs,
So perhaps is a different method in the XCDYouTubeKit, more suitable to my needs?

Cheers!

EDIT-
Just ran the 'Allocations' profile, and besides blocking the UI, the same method also takes a lot of memory.
Calling `[[XCDYouTubeClient defaultClient] getVideoWithIdentifier:_videoID completionHandler:^(XCDYouTubeVideo *video, NSError *error)` for those 25 search results items, 'costed' about 350MB.

**Comment by 0xced** [comment]
I’m looking into it.

**Comment by jlott1** [comment]
I had the same problem, the cause is that NSURLConnection and other methods are operating on main thread runloop when they don't need to since all of this is within an NSOperation.  I was able to get around this by using my own JLURLConnection (https://github.com/jlott1/JLURLConnection) and running this synchronously.  ping me any time and I'd be happy to help.  this is great code otherwise.

**Comment by 0xced** [comment]
Can you please test the `feature/performance` branch? All networking and parsing code is now executed on a background thread.

**Comment by AMI289** [comment]
Haven't made thorough testing,
But it seems to be working great now.

Thank you for your fast reply and extremely fast code adjustments!

You rock Mr. Luthi!

And Mr. Lott, I appreciate your reply, and will to help, as well,
But Mr. Luthi was so extremely fast on solving this, that I didn't had time to try your solution.

Thank you both guys.

**Comment by 0xced** [comment]
Great! This ticket will close automatically when I'll release version 2.2.0.
